tests/nat/front_ends/mcp/test_mcp_debug_routes.py

```diff
@@ -0,0 +1,321 @@
+    worker._setup_debug_endpoints(mcp_server, test_functions)
+    return mcp_server
+
+
+# Streamlined test cases focusing on essential functionality
+class TestDebugRouteBasics:
+    """Test basic functionality of the debug route."""
+
+    @pytest.mark.asyncio
+    async def test_route_exists_and_structure(self, setup_debug_endpoints, test_functions):
+        """Test that the debug route is accessible and returns expected structure."""
+        with TestClient(setup_debug_endpoints.streamable_http_app()) as client:
+            resp = client.get("/debug/tools/list")
+            assert resp.status_code == 200
+            data = resp.json()
+
+            # Check response structure
+            assert "count" in data
+            assert "tools" in data
+            assert "server_name" in data
+            assert data["server_name"] == "Test Server"
+            assert data["count"] == len(test_functions)
+
+            # Check all tools are listed
+            tool_names = {tool["name"] for tool in data["tools"]}
+            expected_names = set(test_functions.keys())
+            assert tool_names == expected_names
+
+
+class TestDetailParameter:
+    """Test the detail parameter behavior."""
+
+    @pytest.mark.parametrize(
+        "detail_param,expected_schema",
+        [
+            ("true", True),
+            ("false", False),
+            ("invalid", True)  # Invalid defaults to True when names specified
+        ])
+    @pytest.mark.asyncio
+    async def test_detail_with_names(self, setup_debug_endpoints, detail_param, expected_schema):
+        """Test detail parameter when tool names are specified."""
+        with TestClient(setup_debug_endpoints.streamable_http_app()) as client:
+            resp = client.get(f"/debug/tools/list?name=regular_tool&detail={detail_param}")
+            data = resp.json()
+
+            if expected_schema:
+                assert "schema" in data["tools"][0]
+                assert data["tools"][0]["schema"] is not None
+            else:
+                assert "schema" not in data["tools"][0]
+
+    @pytest.mark.parametrize(
+        "detail_param,expected_schema",
+        [
+            ("true", True),
+            ("false", False),
+            ("invalid", False)  # Invalid defaults to False when no names
+        ])
+    @pytest.mark.asyncio
+    async def test_detail_without_names(self, setup_debug_endpoints, detail_param, expected_schema):
+        """Test detail parameter when no tool names are specified."""
+        with TestClient(setup_debug_endpoints.streamable_http_app()) as client:
+            resp = client.get(f"/debug/tools/list?detail={detail_param}")
+            data = resp.json()
+
+            for tool in data["tools"]:
+                assert "is_workflow" in tool  # Always present
+                if expected_schema:
+                    assert "schema" in tool
+                else:
+                    assert "schema" not in tool
+
+    @pytest.mark.asyncio
```

Comment:
This should no longer be necessary. Maybe test to verify and remove if thats the case here